OpenRA.Game/Map/MapPreview.cs

```diff
@@ -434,74 +434,57 @@ public void Install(string mapRepositoryUrl, Action onSuccess)
 			}
 
 			var mapInstallPackage = installLocation.Key as IReadWritePackage;
-			new Thread(() =>
+
+			Task.Run(async () =>
 			{
 				// Request the filename from the server
 				// Run in a worker thread to avoid network delays
 				var mapUrl = mapRepositoryUrl + Uid;
-				var mapFilename = string.Empty;
 				try
 				{
-					var request = WebRequest.Create(mapUrl);
-					request.Method = "HEAD";
-					using (var res = request.GetResponse())
+					void OnDownloadProgress(long total, long received, int percentage)
 					{
-						// Map not found
-						if (res.Headers["Content-Disposition"] == null)
-						{
-							innerData.Status = MapStatus.DownloadError;
-							return;
-						}
-
-						mapFilename = res.Headers["Content-Disposition"].Replace("attachment; filename = ", "");
+						DownloadBytes = total;
+						DownloadPercentage = percentage;
 					}
 
-					Action<DownloadProgressChangedEventArgs> onDownloadProgress = i => { DownloadBytes = i.BytesReceived; DownloadPercentage = i.ProgressPercentage; };
-					Action<DownloadDataCompletedEventArgs> onDownloadComplete = i =>
+					var client = new HttpClient();
```

Comment:
This should be using your new `HttpClientFactory`.
```suggestion
					var client = HttpClientFactory.Create();
```